torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -1014,6 +1021,35 @@ struct NCCLTraceBuffer {
     return result;
   }
 
+  // dump pg_status
+  const c10::Dict<c10::IValue, c10::IValue> getPgStatus() {
+    auto all_pg_status = new_dict();
+    for (const auto& [pg_id, status] : all_pg_status_) {
+      auto pg_status = new_dict();
+      pg_status.insert("last_enqueued_collective", status->lastEnqueuedSeq);
+      pg_status.insert("last_started_collective", status->lastStartedSeq);
+      pg_status.insert("last_completed_collective", status->lastCompletedSeq);
+      all_pg_status.insert(std::to_string(pg_id), pg_status);
+    }
+    return all_pg_status;
+  }
+
+  const std::map<std::string, std::map<std::string, std::string>>
+  getPgStatusJson() {
```

Comment:
There is no 'json' about this function? just a map of strings? only seems this is just 3 fields in original pg_status is needed?

Reply:
IIUC, in function `dump_json` it will be converted to json. (Similar things are done in `getPgConfigJson`). For now I only want to include three fields in the final dump, but we can change it if necessary. 

To clarify this does not change scuba logging and we just want to make the struct also available in the ring buffer as well. And since we only keep a pointer so there is no difference no matter how many fields we need in FR?

Reply:
For conversion to JSON the function needs to return `map` instead of `dict`. The JSON conversion library we use (`nlohmann/json`) cannot convert python dicts to JSON natively.
We can maybe rename these functions as `getPgStatusAsMap`?